.github/workflows/terraform.yaml

```diff
@@ -1,28 +1,48 @@
-name: Check the Terraform Module
+name: Check Terraform Module
 
 on:
   workflow_call:
+    paths:
+      - '**.tf'
 
 jobs:
   build:
     name: Terraform Checks and Plans
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout the repository to the runner
+      - name: Checkout Repository
         uses: actions/checkout@v4
 
-      - name: HashiCorp - Setup Terraform
+      - name: Setup Terraform
         uses: hashicorp/setup-terraform@v3
 
-      - name: Terraform Format
-        id: fmt
-        run: terraform fmt -recursive -check
-        continue-on-error: true
-
-      - name: Terraform Init
+      - name: Initialize Terraform Module
         id: init
         run: terraform init
 
-      - name: Terraform Validate
+      - name: Validate Terraform Module
         id: validate
-        run: terraform validate -no-color
\ No newline at end of file
+        run: terraform validate -no-color
+
+      - name: Validate Terraform Formatting
+        run: |
+          set +e 
+
+          echo "CHECK_STATUS=0" >> $GITHUB_ENV
+
+          for changed_file in ${{ steps.pr_files.outputs.added_modified }}; do 
+            echo "Checking terraform fmt on ${changed_file}..."
+
+            if [[ $changed_file == *.tf ]]; then 
+              terraform fmt -check $changed_file
+              FMT_STATUS=$(echo $?)
```

Comment:
you can just use `FMT_STATUS=$?`, or even right to the next line like
```
terraform fmt -check $changed_file
if [[ $? -ne 0 ]]; then
...
```

Reply:
Done